src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/specification/VisitSpecification.kt

```diff
@@ -7,53 +7,53 @@ import jakarta.persistence.criteria.Root
 import org.springframework.data.jpa.domain.Specification
 import uk.gov.justice.digital.hmpps.visitscheduler.model.OutcomeStatus
 import uk.gov.justice.digital.hmpps.visitscheduler.model.VisitFilter
+import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.Visit
 import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.Prison
-import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.OldVisit
 import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.VisitVisitor
 
-class VisitSpecification(private val filter: VisitFilter) : Specification<OldVisit> {
+class VisitSpecification(private val filter: VisitFilter) : Specification<Visit> {
   override fun toPredicate(
-    root: Root<OldVisit>,
+    root: Root<Visit>,
     query: CriteriaQuery<*>,
     criteriaBuilder: CriteriaBuilder,
   ): Predicate? {
     val predicates = mutableListOf<Predicate>()
 
     with(filter) {
       prisonerId?.run {
-        predicates.add(criteriaBuilder.equal(root.get<String>(OldVisit::prisonerId.name), prisonerId))
+        predicates.add(criteriaBuilder.equal(root.get<String>(Visit::prisonerId.name), prisonerId))
       }
 
       prisonCode?.run {
         val innerJoinFromVisitToVisitors =
-          root.join<OldVisit, MutableList<Prison>>(OldVisit::prison.name).get<VisitVisitor>(
+          root.join<Visit, MutableList<Prison>>(Visit::prison.name).get<VisitVisitor>(
             Prison::code.name,
           )
         predicates.add(criteriaBuilder.equal(innerJoinFromVisitToVisitors, prisonCode))
       }
 
       startDateTime?.run {
-        predicates.add(criteriaBuilder.greaterThanOrEqualTo(root.get(OldVisit::visitStart.name), startDateTime))
+        predicates.add(criteriaBuilder.greaterThanOrEqualTo(root.get(Visit::visitStart.name), startDateTime))
```

Comment:
llink to session slot? or replace with date?

Reply:
This code will be removed